cpp/epidemiology/abm/person.cpp

```diff
@@ -66,9 +73,15 @@ void Person::interact(TimeSpan dt, const GlobalInfectionParameters& global_infec
         }
     }
 
-    if (new_state == InfectionState::Infected_Detected || new_state == InfectionState::Infected_Severe || new_state == InfectionState::Infected_Critical) {
+    if (new_state == InfectionState::Infected_Severe || new_state == InfectionState::Infected_Critical) {
         m_quarantine = true;
     }
+    else if (new_state == InfectionState::Infected) {
+        double rand = UniformDistribution<double>::get_instance()();
+        if (rand < global_infection_params.get<TestWhileInfected>()[this->m_age] / dt.days()) {
```

Comment:
doesn't this need to be `p * t` instead of `p / t` (p is probability, t is time step)? Probability per time step is lower if the time step is shorter.

Reply:
Yes, I will change that